Matlab/Matlab.sublime-syntax

```diff
@@ -48,477 +76,792 @@ contexts:
+        - match: \n
+          scope: meta.shell-escape.matlab string.unquoted.matlab
+          pop: true
+    - match: \b\@
+      scope: punctuation.accessor.at.matlab
+    - match: \@
+      scope: keyword.operator.at.matlab
+      push:
+        - match: \(
+          scope: punctuation.section.parens.begin.matlab
+          set:
+            - meta_scope: meta.function.parameters.matlab
+            - include: line-continuation
+            - include: eol-pop
+            - match: '{{identifier}}'
+              scope: variable.parameter.input.matlab
+            - include: separator-comma
+            - match: \)
+              scope: punctuation.section.parens.end.matlab
+              pop: true
+        - match: ''
+          pop: true
+
   transpose:
-    - match: '\b({{id}})(\.?'')'
+    - match: (?:\s*(\.\')|(\'))?
       captures:
-        1 : meta.variable.other.valid.matlab
-        2 : keyword.operator.transpose.matlab
-  transpose_post_parens:
-    - match: '(\.?'')'
-      captures:
-        1 : keyword.operator.transpose.matlab
-      pop: true
-    - match: ''
+        1: keyword.operator.transpose.matlab
+        2: keyword.operator.transpose.matlab
       pop: true
-  unescaped_quote:
-    - match: "'(?=.)"
-      scope: invalid.illegal.unescaped-quote.matlab
-  variable:
-    - match: '\b{{id}}\b'
-      comment: Valid variable. Added meta to disable highlighting
-      scope: meta.variable.other.valid.matlab
-  variable_assignment:
-    - match: '=\s*\.{0,2}\s*;?\s*$\n?'
-      comment: Incomplete variable assignment.
-      scope: invalid.illegal.incomplete-variable-assignment.matlab
-  variable_invalid:
+
+  accessor-dot:
+    - match: (?<=\S)\.
+      scope: punctuation.accessor.dot.matlab
+
+  terminator:
+    - match: \;
+      scope: punctuation.terminator.matlab
+
+  separator-comma:
+    - match: \,
+      scope: punctuation.separator.sequence.matlab
+
+  separator-semicolon:
+    - match: \;
+      scope: punctuation.separator.sequence.matlab
+
+###[ BUILTIN CONSTANTS, VARIABLES AND DATA TYPES ]############################
+
+  # Functions that are usually used without parameters and return a constant
+  builtin-constants:
+    - match: \b(true|false|eps|pi|Inf|NaN|NaT|flintmax|intmax|intmin|realmax|realmin|namelengthmax)\b
+      scope: constant.language.matlab
+      push: transpose
+
+  builtin-variables:
+    - match: \b(ans|nargin|nargout|varargin|varargout)\b
+      scope: variable.language.matlab
+      push: transpose
+
+  builtin-types:
+    - match: \b(?:categorical|cell|char|complex|double|int8|int16|int32|int64|logical|single|string|struct|table|timeseries|timetable|uint8|uint16|uint32|uint64)\b(?!\()
+      scope: storage.type.matlab
+
+  last-index:
+    - match: \bend\b
+      scope: variable.language.matlab
+      push: transpose
+
+  argument-placeholder:
+    - match: \~(?![A-Za-z])
+      scope: variable.language.blank.matlab
```

Comment:
It appears several languages such as Python, Java or Scala have such kinds of special variables to denote unrequired or arbritary variables. I'd find it suitable to have a common scope for all of them. While `blank` fits quite well for that use case, I wonder whether `anonymous` may be an alternative. The reason for it is scoping lambda functions `anonymous` as well, so reusing this subscope name for unnamed variables seems logical.

Thoughts?

Reply:
There was a question/discussion for this in https://github.com/SublimeText/ScopeNamingGuidelines/issues/5, and I mentioned that `variable.language.blank` from the Go syntax might be a good solution. A search on this repo revealed that you used `variable.language.anonymous` in Erlang, so I adjusted this scope here in 12a1469. Would definitely be useful to have a common convention for it and unified scopes in the default syntaxes.

Maybe worth to mention that in the [Matlab docs](https://www.mathworks.com/help/matlab/matlab_prog/ignore-function-outputs.html) it is called "tilde operator", but I don't find that wording a good choice, because `~` can also be used as the negation operator.